### PR TITLE
chore(gatsby): upgrade dependencies

### PR DESCRIPTION
**Pull request by alexandernanberg**
This PR upgrades a bunch of dependencies of the `gatsby` package. There are still some outdated ones but I think this is a good start! From what I can tell we were not affected by any of the breaking changes (mostly node version bumps).

I haven't run `yarn upgrade` to update all the packages in the lockfile (can add that to the PR if you'd like)

Although the tests are passing –  we probably want to do some testing with "real" projects, such as the starter kits.

### CODE REVIEW

**Comment by pieh** [comment]
Did you test if it works with updated dependencies? `gatsby-dev` doesn't handle dependency changes so one way to do that would be use `npm/yarn pack` to create tarball of gatsby package with changed dependencies and install that tarball in test site

**Comment by pieh** [comment]
@alexandernanberg I tried this and get same result when using `yarn add` but it works with `npm install` (for some reason) - I didn't investigate yet why it's happening :(

File: packages/gatsby/package.json [hunk 1]
```diff
@@ -182,7 +180,6 @@
     "clean-test-bundles": "find test/ -type f -name bundle.js* -exec rm -rf {} +",
     "prebuild": "rimraf dist && rimraf cache-dir/commonjs",
     "prepare": "cross-env NODE_ENV=production npm run build",
-    "test-coverage": "node_modules/.bin/nyc --reporter=lcov --reporter=text npm test",
```
Comment (alexandernanberg) [inline packages/gatsby/package.json]: This isn't used right? The command itself doesn't do anything

**Comment by alexandernanberg** [comment]
@pieh Ah I see, npm worked for me as well. I guess there is some problems with `pack` + `workspaces`, because I think `pack` needs a `yarn.lock` file to work.

I've tested in `gatsby-starter-blog`, `gatsby-starter-default` and did not experience any problems. Do we want to test in some larger projects also?

**Comment by alexandernanberg** [comment]
@pieh Merged with master and I've now tested with a somewhat larger project I'm currently working on and didn't run into any problems.

**Pull request closed without merging**